### PR TITLE
Issue #64 : Implement backup mode

### PR DESCRIPTION
**Pull request by Wesam18**

### CODE REVIEW

File: Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb [hunk 1]
```diff
@@ -71,11 +72,30 @@ def lock_blob(blob_names)
     return NIL
   end # def lock_blob
   
+  def archive_blob(blob_names)
+    real_blob_names = blob_names.select { |name| !name.end_with?(".bak") }
+    real_blob_names.each do |blob_name|
+      new_blob_name = "#{blob_name}-#{Time.now.to_f}.bak"
+      @azure_blob.copy_blob(@container,new_blob_name, @container, blob_name)  
+      @azure_blob.delete_blob(@container, blob_name)
```
Comment (brahmnes) [inline Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: This is an interesting idea, but it wouldn't work if you have multiple readers which happen to be reading the same blob. I think there still needs to be some locking mechanism using the acquire method. If you can figure out a fusion of the archive and lock approach that works in the multi-reader case, we can make that the default.

**Comment by Wesam18** [comment]
I implemented your suggest change

**Comment by simonwh** [comment]
This is good, can we have this merged?

**Comment by brahmnes** [comment]
The solution is still not ideal because while it's holding the lock, it will prevent writers from writing the blob.
Having said that, I still don't have a better solution in mind and I will keep thinking about it. In the meantime, I can accept this merge and keep this a non-default option.

**Pull request merged**